avm/ptn/azd/ml-hub-dependencies/main.bicep

```diff
@@ -0,0 +1,475 @@
+@allowed([
+  'default'
+  'highDensity'
+])
+param hostingMode string = 'default'
+
+@description('Optional. Network specific rules that determine how the Azure Cognitive Search service may be reached.')
+param networkRuleSet object = {
+  bypass: 'None'
+  ipRules: []
+}
+
+@description('Optional. The number of partitions in the search service; if specified, it can be 1, 2, 3, 4, 6, or 12. Values greater than 1 are only valid for standard SKUs. For \'standard3\' services with hostingMode set to \'highDensity\', the allowed values are between 1 and 3.')
+@minValue(1)
+@maxValue(12)
+param partitionCount int = 1
+
+@description('Optional. This value can be set to \'Enabled\' to avoid breaking changes on existing customer resources and templates. If set to \'Disabled\', traffic over public interface is not allowed, and private endpoint connections would be the exclusive access method.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param searchServicePublicNetworkAccess string = 'Enabled'
+
+@description('Optional. The number of replicas in the search service. If specified, it must be a value between 1 and 12 inclusive for standard SKUs or between 1 and 3 inclusive for basic SKU.')
+@minValue(1)
+@maxValue(12)
+param replicaCount int = 1
+
+@allowed([
+  'disabled'
+  'free'
+  'standard'
+])
+@description('Optional. Sets options that control the availability of semantic search. This configuration is only possible for certain search SKUs in certain locations.')
+param semanticSearch string = 'disabled'
+
+@description('Optional. Defines the SKU of an Azure Cognitive Search Service, which determines price tier and capacity limits.')
+@allowed([
+  'basic'
+  'free'
+  'standard'
+  'standard2'
+  'standard3'
+  'storage_optimized_l1'
+  'storage_optimized_l2'
+])
+param searchServiceSku string = 'standard'
+
+@description('Conditional. The managed identity definition for this resource. Required if `assignRbacRole` is `true` and `managedIdentityName` is `null`.')
+param managedIdentities managedIdentitiesType?
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.azd-mlhubdependencies.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module keyVault 'br/public:avm/res/key-vault/vault:0.7.1' = {
```

Comment:
Note that all references to AVM modules inside your module should pass the `enableTelemetry` value through

Reply:
This should not have ben resolved.
Please note @Menghua1, I'm talking about a passthru - not a condition of the entire deployment.

You'd always want to deployment, independent of the telemetry. What I'm asking for is adding `enableTelemetry: enableTelemetry` to to params block of each module reference. Please do that and remove the condition. And don't resolve the comments, but just reply that you've completed the ask 😉